behaviour/graql/language/match.feature

```diff
@@ -24,7 +24,65 @@ Feature: Graql Match Clause
     Given transaction is initialised
     Given the integrity is validated
 
-  Scenario: Disjunctions return the union of composing query statements
+  Scenario: `type` matches only the specified type, and does not match subtypes
+
+  Scenario: `sub` matches the specified type and all its subtypes
+
+  # TODO: ????
+  Scenario: `sub!` matches ??? (?)
+
+  # TODO: ensure there are types that don't have the specified attribute, as well, so we know it's a precise match
+  Scenario: `has` matches types that have the specified attribute type
+
+  Scenario: `has` matches types that have only a subtype of the specified attribute type (?)
+
+  Scenario: `plays` matches types that can play the specified role
+
+  Scenario: `key` matches types that have the specified attribute type as a key (?)
+
+  Scenario: `has` matches types that have the specified attribute type, even if they use it as a key (?)
+
+  Scenario: `relates` matches types that have the specified roleplayer, regardless of their role
+
+  Scenario: `relates` with `as` matches only types that have the specified roleplayer playing the specified role
+
+  Scenario: `relates` matches/does not match types that block the specified roleplayer with `as` (?)
+
+  Scenario: `isa` matches things of the specified type and all its subtypes
+
+  Scenario: `isa!` matches only things of the specified type, and does not match subtypes
+
+  Scenario: when an attribute type is specified but no value, `has` matches things that have any instance of the specified attribute (?)
+
+  Scenario: when a value is specified but no type, `has` matches things that have any kind of attribute with that value
+
+  Scenario: when an attribute instance is fully specified, `has` matches all of its owners
+
+  Scenario: `has` with an instance fully specified matches all of its owners, even if they own other instances of the same attribute
+
+  Scenario: `contains` matches strings that contain the specified substring
+
+  Scenario: `like` matches strings that match the specified regex
+
+  Scenario: `has $attr > $x` matches owners of any instance `$y` of `$attr` where `$y > $x`
+
+  Scenario: `has $attr < $x` matches owners of any instance `$y` of `$attr` where `$y < $x`
+
+  Scenario: `has $attr !== $x` matches owners of any instance `$y` of `$attr` where `$y !== $x`
+
+  Scenario: `has $attr > $x` matches owners of any instance `$y` of `$attr` where `$y > $x` even if they also own instance `$z` where `$z < $x`
+
+  Scenario: `id` matches the instance with the specified internal id
+
+  Scenario: disjunctions return the union of composing query statements
+
+  Scenario: a negation matches if the negated block has no matches
+
+  Scenario: a negation does not match if the negated block has any matches
+
+  Scenario: `has` matches attribute ownerships inferred from rules
+
+  Scenario: `isa` matches relations inferred from rules
```

Comment:
I'm wondering if we want to have a whole separate file for testing reasoner capabilities, because there are some more:
1. appending role players
2. adding a "second" supertype of an instance (this is a sneaky reasoning thing we can do, not documented anywhere ;), maybe we can get kasper who wrote the entire reasoning engine to review the set of reasoning scenarios in the future)
3. creating new attribute and an ownership thereof in the `then` of the rule..


there are more, i just think that is a good  separate scope to have a new file for.

what do you think?

Reply:
there's even a `reasoning` package in verification/behaviour already i think~

Reply:
Ok I will run the name `reasoner/inference.feature` past Haikal.

Reply:
We're thinking `resolution.feature` (which might change in the future as more tests are added)